File: solutions/day-10/nodejs/puzzle-2.js
#!/usr/bin/env node
'use strict';

const fs = require('fs');
const path = require('path');
const { KnotHash } = require('./knot-hash');

const input = fs.readFileSync(path.resolve(__dirname, '../input.txt'), 'utf-8');

console.log(KnotHash.Hash(input));

File: solutions/day-01/python/puzzle-2.py
#!/usr/bin/env python3
import os

inputPath = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../input.txt"))

file = open(inputPath, 'r')
input = file.read()
file.close()

sum = 0
offset = int(len(input) / 2)

for index in range(0, len(input) - 1):
    target = (index + offset) % len(input)
    if input[index] == input[target]:
        sum += int(input[index])

print(sum)